### PR TITLE
Use uv as python package manager instead of poetry in docker

### PR DESCRIPTION
**Pull request by signed-long**
**Description**

Uses uv to install python dependencies instead of poetry if "uv" is passed into `install_python_dependencies.sh`. 

Uv's requirements.txt files are platform specific so this is just for the docker build as of now. Also, uv doesn't have dependencies groups like poetry for dev dependencies, so this installs all deps. 

I can add requirements.txt files to support mac builds and only non-dev deps too if the bounty #32522 gets locked to me, but with this PR the CI speed up may be worth merging. Seeing setup-with-retry with no cache hits taking 4min from 7min. 

**Verification**

 Builds and passes tests in CI.

### CODE REVIEW

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Lack of locking is a bit annoying, but the speedup is likely worth it. Let's do this:
* remove `requirements.docker.txt`
* just do `uv pip install .` in the Dockerfile (i.e. remove the install_python_dependencies.sh usage

Then, we can merge this, and I'll make a follow up to remove `install_python_dependencies.sh`. We really shouldn't be changing people's pip versions or anything weird like that with our weird setup script.

**Comment by adeebshihadeh** [comment]
Here's my follow up to this: #32541

File: pyproject.toml [hunk 1]
```diff
@@ -1,12 +1,137 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
```
Comment (adeebshihadeh) [inline pyproject.toml:3]: can you revert the changes here and to install_python_dependencies? I want to keep the diff small on this PR in case it makes sense to revert (if the locking actually becomes a problem)

Comment (signed-long) [reply pyproject.toml]: Had to make some change to get the poetry pyproject.toml to work with uv, but it's a smaller diff than this was

**Comment by signed-long** [comment]
The 7min time was from my memory playing with setup-retry last week, actual speed up building image with no cache in gha: 
- uv 27s:  https://github.com/signed-long/openpilot/actions/runs/9245895114/job/25433013590#step:3:4871 
- poetry 59s: https://github.com/signed-long/openpilot/actions/runs/9245895114/job/25433013642#step:3:5078

**Comment by adeebshihadeh** [comment]
Unfortunately, the locking has already become an issue. Our weekly lockfile CI job just ran, and a couple different things broke (https://github.com/commaai/openpilot/pull/32550). It would've been bad if those just randomly broke on master instead of that PR.

I'm closing it, but the bounty  is yours since all the work was done.
